http/cors.go

```diff
@@ -0,0 +1,32 @@
+package http
+
+import (
+	"net/http"
+
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/vault"
+)
+
+func wrapCORSHandler(h http.Handler, core *vault.Core) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		corsConf := core.CORSConfig()
```

Comment:
@jefferai CORS is a confusing subject. I've learned more about it than I ever wanted, since I started working on this PR. Let's see if I can help alleviate your confusion. 

The browser will **not send** a preflight request to Vault (and any cross-origin resource) for the following requests:

- `GET`
- `HEAD`
- `POST` if the content-type of the `POST` request is `text/plain`, `application/x-www-form-urlencoded`, or `multipart/form-data`.

For anything else the browser **will send** a preflight request to Vault (and any cross-origin resource).

Regardless of the browser's need for a preflight request the browser will expect an `Access-Control-Allow-Origin` header on the response from Vault and any cross-origin resource.

This is a pretty good explanation for CORS: http://restlet.com/blog/2015/12/15/understanding-and-using-cors/